ratschlab_common/io/sparse_df.py

```diff
@@ -0,0 +1,122 @@
+import pandas as pd
+import numpy as np
+from pandas.api.types import is_sparse
+from scipy import sparse
+import tables
+
+'''
+This module exposes two functions: to_hdf and read_hdf, which usage
+is to read and write sparse Pandas DataFrames into and from hdf file.
+'''
+
+
+def read_hdf(fpath):
+    '''
+    read a hdf file into a sparse Pandas DataFrame that was previously saved with the to_hdf function of this module.
+
+    :param fpath: location of the hdf file
+    :return: an instance of Pandas DataFrame
+    '''
+    with tables.open_file(fpath, mode='r') as f:
+        sparse_group = f.get_node("/data/sparse")
+        sparse_m = _read_sparse_m_coo(sparse_group)
+        col_names = f.get_node("/col_names")
+        s_col_names = f.get_node("/data/sparse/sparse_col_names")
```

Comment:
the elements of `f.get_node("/col_names") ` and `f.get_node("/data/sparse/sparse_col_names")` are byte arrays, you would need to convert them to strings like `[c.decode('UTF-8') for c in f.get_node("/col_names") ]`. Otherwise there are issues further down in `ret_df = ret_df[col_names]`